embassy-nrf/src/pwm.rs

```diff
@@ -24,7 +24,7 @@ pub struct SimplePwm<'d, T: Instance> {
 
 /// SequencePwm allows you to offload the updating of a sequence of duty cycles
 /// to up to four channels, as well as repeat that sequence n times.
-pub struct SequencePwm<'d, T: Instance> {
+pub struct SequencePwm<'d, T: Instance, const N: usize> {
```

Comment:
The generic here will caues rustc to duplicate ALL the `SequencePwm` code if you instantiate it with different lengths, which is quite likely.

Also it doesn't allow playing sequences with length decided at runtime.

I would instead make it take a `&mut [u16]`